assignment7.txt

```diff
@@ -0,0 +1,23 @@
+Julia Tan
+
+1. The order of operations of prefix ++ and * is the same. It is read from right to left. The order of operations of postfix ++ is higher than both * and prefix ++. It is read from left to right.
+
+++*p would be read as ++(*p). (dereferences p then increments it)
+*p++ would be read as *(p++). (increments p then dereferences it)
+*++p would be read as *(p++) as well. (incremeents p then dereferences it)
+
+2. No it is not guaranteed, as sometimes expressions are read left to right, and other times from right to left, as seen on the example above.
+
+3. Pointers are useful for handling large amounts of data efficiently. They are also used to access variables in other functions without using global variables, as global variables can be confusing and difficult to debug. Other times, pointers are essential, such as in complex data structures.
+
+4.
+4.1 char * (char * are strings)
+4.2 invalid (if "xyz" is an array, you cannot subtract letters in each element)
```

Comment:
This is actually valid! "xyz" is just an array of characters, so the "xyz"[1] is just accessing "y". Then you just subtract 'y' from it to get 0. This is because, if you can recall, a char is just a value mapped to a character!